flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -195,9 +196,24 @@ public FlowCheckResponse getFlowState(String flowId) {
         flowCheckResponse.setFlowId(flowId);
         flowCheckResponse.setHasActiveFlow(!completed("Flow", flowId, List.of(), allByFlowIdOrderByCreatedDesc));
         flowCheckResponse.setLatestFlowFinalizedAndFailed(isFlowInFailedState(allByFlowIdOrderByCreatedDesc, failHandledEvents));
+        setStateEventFlowTypeOnFlowCheckResponse(flowCheckResponse, allByFlowIdOrderByCreatedDesc,"flow", flowId);
         return flowCheckResponse;
     }
 
+    public void setStateEventFlowTypeOnFlowCheckResponse(FlowCheckResponse flowCheckResponse, List<FlowLogWithoutPayload> relatedFlowLogs, String type, String id) {
+        if (flowCheckResponse.getHasActiveFlow() && !relatedFlowLogs.isEmpty()) {
+            LOGGER.info("Setting current state, next event, and flow type into flow check response using {} {}", type, id);
+            FlowLogWithoutPayload firstFlowLog = relatedFlowLogs.get(0);
```

Comment:
This is rather the `latestFlowLog`, not the first.

Reply:
@lturcsanyi Updated that. TYSM!!